src/main/java/org/killbill/billing/plugin/helloworld/HelloWorldInvoicePluginApi.java

```diff
@@ -0,0 +1,151 @@
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.plugin.helloworld;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import org.killbill.billing.account.api.Account;
+import org.killbill.billing.invoice.api.Invoice;
+import org.killbill.billing.invoice.api.InvoiceItem;
+import org.killbill.billing.invoice.api.InvoiceItemType;
+import org.killbill.billing.notification.plugin.api.ExtBusEvent;
+import org.killbill.billing.osgi.libs.killbill.OSGIConfigPropertiesService;
+import org.killbill.billing.osgi.libs.killbill.OSGIKillbillAPI;
+import org.killbill.billing.osgi.libs.killbill.OSGIKillbillEventDispatcher.OSGIKillbillEventHandler;
+import org.killbill.billing.payment.api.PluginProperty;
+import org.killbill.billing.plugin.api.invoice.PluginInvoiceItem;
+import org.killbill.billing.plugin.api.invoice.PluginInvoicePluginApi;
+import org.killbill.billing.util.callcontext.CallContext;
+import org.killbill.billing.util.callcontext.TenantContext;
+import org.killbill.clock.Clock;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+class HelloWorldInvoicePluginApi extends PluginInvoicePluginApi implements OSGIKillbillEventHandler {
+
+	public HelloWorldInvoicePluginApi(OSGIKillbillAPI killbillAPI, OSGIConfigPropertiesService configProperties,
+			Clock clock) {
+		super(killbillAPI, configProperties, clock);
+	}
+
+	/**
+	 * Returns additional invoice items to be added to invoice
+	 * <p>
+	 * This method produces two types of invoice items a. Tax Item on Invoice Item
+	 * b. Adjustment Item on only the very first Historical Invoice Tax Item
+	 * automatically ( just for Demo purpose )
+	 *
+	 * @param newInvoice The invoice that is being created.
+	 * @param dryRun     Whether it is dryRun or not
+	 * @param properties Any user-specified plugin properties, coming straight out
+	 *                   of the API request that has triggered this code to run.
+	 * @param callCtx    The context in which this code is running.
+	 * @return A new immutable list of new tax items, or adjustments on existing tax
+	 *         items.
+	 */
+	@Override
+	public List<InvoiceItem> getAdditionalInvoiceItems(final Invoice newInvoice, final boolean dryRun,
+			final Iterable<PluginProperty> properties, final CallContext callContext) {
+
+		UUID accountId = newInvoice.getAccountId();
```

Comment:
The code does not look to be formatted as per Kill Bill coding standards here and elsewhere.  As per Kill Bill coding standards, a variable is declared as `final`, whereas `accountId`  (and other variables in the code) are not declared as `final`.